### PR TITLE
Move `transport` and `api_version` parameters into kwargs

### PR DESCRIPTION
**Pull request by chlowell**

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/_shared/async_client_base.py [hunk 1]
```diff
@@ -75,18 +68,16 @@ def __init__(
...
 
-        if api_version is None:
-            api_version = KeyVaultClient.DEFAULT_API_VERSION
-
-        config = self._create_config(credential, api_version=api_version, **kwargs)
+        config = self._create_config(credential, **kwargs)
+        transport = kwargs.pop("transport", None)
         pipeline = kwargs.pop("pipeline", None) or self._build_pipeline(config, transport=transport, **kwargs)
-        self._client = KeyVaultClient(credential, api_version=api_version, pipeline=pipeline, aio=True)
+        self._client = KeyVaultClient(credential, pipeline=pipeline, aio=True)
```
Comment (johanste) [inline sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/_shared/async_client_base.py]: Was this change intentional?

Comment (chlowell) [reply sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/_shared/async_client_base.py]: This is half the intended change, need to pass kwargs along as well. Thanks for catching it. I opened #7778 to track testing API versions.

File: sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/aio/client.py [hunk 2]
```diff
@@ -28,8 +28,17 @@
...
-    """"CertificateClient defines a high level interface for
-    managing certificates in the specified vault.
+    """A high-level asynchronous interface for managing a vault's certificates.
+
+    :param str vault_endpoint: URL of the vault the client will access
+    :param credential: An object which can provide an access token for the vault, such as a credential from
+        :mod:`azure.identity.aio`
+
+    Keyword arguments
+        - **api_version**: version of the Key Vault API to use. Defaults to the most recent.
```
Comment (KieranBrantnerMagee) [inline sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/aio/client.py]: str here?  We're not super consistent in specifying type for keyword args, I know in servicebus we try to specify across the board; but obviously prefer whichever makes you more consistent with your scope.
